### PR TITLE
Cw first non repeating letter

### PR DESCRIPTION
**Pull request by aepaysinger**
this was on the CW youre a square branch. so i removed it from there and put it on the right branch.

### CODE REVIEW

File: code_challenges/code_wars/first_non_repeating_letter.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+def first_non_repeating_letter(word):
+    word = list(word)
+    for i, letter in enumerate(word):
```
Comment (W-Ely) [inline code_challenges/code_wars/first_non_repeating_letter.py:3]: a string is both enumerate-able and sliceable so don't think you need to convert it to a list first

File: code_challenges/code_wars/first_non_repeating_letter.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        if letter in word[i + 1 :: 1] or letter in word[0:i]:
+            print(letter, word[i + 1 :: 1], word[0:i])
+            continue
```
Comment (W-Ely) [inline code_challenges/code_wars/first_non_repeating_letter.py:6]: Do you remember what I said about if blocks that do nothing?

**Review by W-Ely — APPROVED** [review]
Is there a readme?  What's the time complexity of this solution?

How might you reduce the time complexity here?

File: code_challenges/code_wars/first_non_repeating_letter.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from collections import Counter
+
+
+def first_non_repeating_letter(word):
+    for letter in word:
+        if Counter(word)[letter] == 1:
+            return letter
```
Comment (W-Ely) [inline code_challenges/code_wars/first_non_repeating_letter.py:7]: What does `Counter(word)` do?  Does the case of the import tell you what kind of Python thing Counter is?

[docs: counter](https://docs.python.org/3/library/collections.html#collections.Counter)

Use counter but don't create a new one for every letter, it's an object so reuse the instance you create a single time.

Comment (aepaysinger) [reply code_challenges/code_wars/first_non_repeating_letter.py]: i think i fixed it. creating a variable, and then using that.

**Review by W-Ely — APPROVED** [review]

**Pull request merged**